src/iOS/toga_iOS/widgets/switch.py

```diff
@@ -0,0 +1,44 @@
+from rubicon.objc import objc_method
+
+from toga.interface import Switch as SwitchInterface
+
+from .base import WidgetMixin
+from ..libs import *
+
+
+class TogaSwitch(UISwitch):
+    @objc_method
+    def onPress_(self, obj) -> None:
+        if self._interface.on_press:
+            self._interface.on_press(self._interface)
+
+
+class Switch(SwitchInterface, WidgetMixin):
+    def __init__(self, label, id=None, on_press=None, style=None, state=False):
+        super().__init__(label, id=id, style=style, on_press=on_press, state=state)
+        self._create()
+
+    def create(self):
+        self._impl = TogaSwitch.alloc().init()
+        self._impl._interface = self
+
+        self._impl.addTarget_action_forControlEvents_(self._impl, get_selector('onPress:'), UIControlEventValueChanged)
```

Comment:
Don't get me wrong, the inconsistency is also the thing that bugs me most. For me it's just a manner of how to add a label not if. I think the switch should always come with a label. This is why I would like to put it into a UITable​View​Cell with a label. This way, if someone wants to make there settings independent of the toga settings API, they can still use it and we can still use it in a UITableView. But I also see the point that there is not much use for a switch element outside of settings. I envisage it to be there for someone who wants to use a switch in a non standard way but be rarely used because the settings api is hopefully so dope that nobody wants to build there settings by hand.